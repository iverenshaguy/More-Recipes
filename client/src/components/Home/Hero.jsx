import React from 'react';
import { InputGroup, InputGroupButton, Button, Form, Label, Input } from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';

const Hero = props => (
  <div className="container-fluid" id="hero-image">
    <div className="row mx-auto" id="hero-image-text">
      <div className="col align-self-center">
        <div className="row justify-content-center align-items-center mb-3">
          <div className="col-12">
            <h3>Search for a Recipe</h3>
          </div>
          <div className="col-8 py-4">
            <Form id="search-form">
              <InputGroup size="lg">
                <Label className="sr-only" for="search">
                  Search for Recipes
                </Label>
                <Input id="search" placeholder="Curried Chicken Gravy..." />
                <InputGroupButton>
                  <Button type="submit" className="search-btn">
                    <FontAwesome name="search" />
                  </Button>
                </InputGroupButton>
              </InputGroup>
            </Form>
          </div>
        </div>
        <Button
          onClick={() => props.toggle('addRecipeModal')}
          className="btn-default btn-lg"
          id="home-add-recipe-btn"
          title="New Recipe"
        >
          Add a New Recipe
        </Button>
      </div>
    </div>
  </div>
);

Hero.propTypes = {
  toggle: PropTypes.func.isRequired,
};

export default Hero;
