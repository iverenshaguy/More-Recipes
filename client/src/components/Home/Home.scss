@import '../../assets/scss/modules';

.home {
  .navbar {
    position: relative;
  }

  #hero-image {
    height: 45vh;
    max-width: 100%;
    position: relative;

    button {
      background-color: $light-green;
      border: 1px solid $light-green;
      color: $white;
      padding: 0.75rem 1.5rem;

      &:hover,
      &:focus,
      &:active {
        background-color: $green;
        border: 1px solid $green;
        color: $white;
      }
    }
  }

  #hero-image-text {
    color: $white;
    padding-top: 5vh;
    text-align: center;
    width: 100%;
  }

  #hero-search-form {
    margin-top: 2rem;
  }
}

#home-add-recipe-btn {
  border-radius: 0;
  box-shadow: 0;
  outline: none;

  &:hover,
  &:focus,
  &:active {
    cursor: pointer;
  }
}
