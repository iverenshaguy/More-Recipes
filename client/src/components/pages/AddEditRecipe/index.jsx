import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Form from '../../shared/Forms';
import { urlMatchPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @class AddEditRecipe
 * @extends Component
 * @returns {component} AddEditRecipe
 */
class AddEditRecipe extends Component {
  static propTypes = {
    ...urlMatchPropTypes,
    type: PropTypes.string,
    uploading: PropTypes.bool.isRequired,
    submitting: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    type: 'addRecipe'
  }

  /**
   * @memberof AddEditRecipe
   * @returns {component} AddEditRecipe
   */
  render() {
    const meta = {
      title: this.props.type === 'addRecipe' ? 'Add a New Recipe' : 'Edit Recipe',
      btnText: this.props.type === 'addRecipe' ? 'Add Recipe' : 'Edit Recipe'
    };

    return (
      <div className="card-form-body">
        <div className="container">
          <div className="my-5" id="card-form-wrapper">
            <div className="p-xs-2 pt-4 pb-3" id="login-signup-div">
              <Form {...this.props} type={this.props.type} meta={meta} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  submitting: state.singleRecipe.recipe.adding,
  uploading: state.uploadImage.uploading,
  submitError: state.singleRecipe.recipe.error
});

export { AddEditRecipe as AddEditRecipeComponent };
export default connect(mapStateToProps)(AddEditRecipe);
