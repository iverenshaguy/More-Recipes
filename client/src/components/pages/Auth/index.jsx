import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { Row, Col } from 'reactstrap';
import Form from '../../shared/Forms';
import setCurrentLocation from '../../../actions/location';
import { authPropTypes } from '../../../helpers/proptypes';
import './Auth.scss';

/**
 * @exports
 * @class Auth
 * @extends Component
 * @returns {component} Auth
 */
class Auth extends Component {
  static propTypes = {
    ...authPropTypes,
    type: PropTypes.string.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  /**
   * @memberof LoginForm
   * @returns {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('auth'));
  }

  /**
   * @memberof Auth
   * @returns {component} Auth
   */
  render() {
    const { from } = this.props.location.state ? this.props.location.state : { from: { pathname: '/' } };

    if (this.props.isAuthenticated) {
      return <Redirect to={from} />;
    }

    const signupMeta = {
      title: 'Register for a New Account',
      btnText: 'SIGN UP',
      extra: <p className="text-center">Already have an account, <Link to="/login">Log in here</Link>.</p>
    };

    const loginMeta = {
      title: 'Sign In to Your Account',
      btnText: 'SIGN IN',
      extra: <p className="text-center">Dont have an account, <Link to="/signup">Sign up here</Link>.</p>
    };

    return (
      <div className="login-signup-body">
        <div className="container" id="login-signup-container">
          <Row className="my-5" id="login-signup-wrapper">
            <Col className="p-xs-2 p-lg-4" id="login-signup-div">
              {this.props.type === 'signup' ?
                <Form {...this.props} type="signup" meta={signupMeta} /> :
                <Form {...this.props} type="login" meta={loginMeta} />}
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

export { Auth as AuthComponent };

const mapStateToProps = state => ({
  submitting: state.auth.loading,
  isAuthenticated: state.auth.isAuthenticated,
  submitError: state.auth.error
});

export default connect(mapStateToProps)(Auth);
