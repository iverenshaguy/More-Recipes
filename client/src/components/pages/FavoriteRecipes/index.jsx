import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { push } from 'react-router-redux';
import FontAwesome from 'react-fontawesome';
import RecipeItems from '../../shared/RecipeItems';
import { MiniPreLoader } from '../../shared/PreLoader';
import { fetchFavoriteRecipes } from '../../../actions/recipes';
import { userPropTypes, multiRecipePropTypes, urlMatchPropTypes } from '../../../helpers/proptypes';
import './FavoriteRecipes.scss';

/**
 * @exports
 * @class FavoriteRecipes
 * @extends Component
 * @classdesc Returns User FavoriteRecipes Page
 * @returns {JSX} User FavoriteRecipes
 */
class FavoriteRecipes extends Component {
  static propTypes = {
    ...userPropTypes,
    ...urlMatchPropTypes,
    ...multiRecipePropTypes,
    isFetching: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired
  }

  /**
   * @memberof FavoriteRecipes
   * @constructor
   * @returns {JSX} FavoriteRecipes
   */
  constructor() {
    super();

    this.state = {
      limit: 5,
      currentPage: 1,
      activeTab: 'recipes'
    };

    this.toggleTab = this.toggleTab.bind(this);
    this.handlePageChange = this.handlePageChange.bind(this);
    // this.showModal = this.showModal.bind(this);
  }

  /**
   * @memberof FavoriteRecipes
   * @returns {nothing} returns nothing
   */
  componentWillMount() {
    const { match, user, dispatch } = this.props;

    if ((match.path).includes('categories')) {
      dispatch(push(`/${user.username}/favorites/categories`));

      this.setState({
        activeTab: 'categories'
      });
    } else if (match.params.username !== user.username) {
      // redirect user to real username if a new uer logins after an old user
      dispatch(push(`/${user.username}/favorites`));
    }

    dispatch(fetchFavoriteRecipes(this.props.user.id, this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetchFavoriteRecipes(this.props.user.id, page, this.state.limit)));
  }

  // /**
  //  * @memberof FavoriteRecipes
  //  * @param {string} modal - modal to show
  //  * @returns {nothing} Returns nothing
  //  */
  // showModal(modal) {
  //   this.props.dispatch(toggleModal(modal));
  // }

  /**
   * @memberof FavoriteRecipes
   * @param {object} e - event
   * @param {string} activeTab - tab
   * @returns {nothing} Returns nothing
   */
  toggleTab(e, activeTab) {
    e.preventDefault();
    const { user } = this.props;
    const location = activeTab === 'recipes' ? `/${user.username}/favorites` : `/${user.username}/favorites/categories`;

    this.setState({ activeTab });
    this.props.dispatch(push(location));
  }

  /**
   * @memberof FavoriteRecipes
   * @returns {JSX} User FavoriteRecipes
   */
  render() {
    const { activeTab } = this.state;

    const {
      isFetching, recipes, metadata, user
    } = this.props;

    const recipesTab = classNames({
      'd-inline': true,
      'recipes-tab': true,
      'text-center': true,
      active: activeTab === 'recipes'
    });

    const categoriesTab = classNames({
      'd-inline': true,
      'recipes-tab': true,
      'text-center': true,
      active: activeTab === 'categories'
    });

    return (
      <div className="favorite-recipes pb-4">
        <h3 className="text-center pt-5">FAVORITE RECIPES</h3>
        <div className="profile-toggle-wrapper text-center mt-3">
          <Link to={`/${user.username}`} className="back-link"><FontAwesome name="arrow-left" tag="i" /> BACK TO PROFILE</Link>
          <a className={recipesTab} href="#recipes" onClick={e => this.toggleTab(e, 'recipes')}>
            <i className="aria-hidden flaticon flaticon-flat-plate-with-hot-food-from-side-view" />
            <span className="d-none d-sm-inline">&nbsp;&nbsp;RECIPES</span>
          </a>
          <a className={categoriesTab} href="#categories" onClick={e => this.toggleTab(e, 'categories')}>
            <FontAwesome name="folder" size="lg" tag="i" />
            <span className="d-none d-sm-inline">&nbsp;&nbsp;CATEGORIES</span>
          </a>
        </div>
        <div className="container-fluid text-center user-profile-recipe-cards-wrapper mb-5 mt-1">
          {isFetching &&
            <div className="justify-content-center py-5">
              <MiniPreLoader />
            </div>}
          {!isFetching &&
            <div className="row justify-content-center user-profile-recipe-cards px-5 px-sm-5 px-md-0 mt-0">
              {activeTab === 'recipes' && (
                <RecipeItems
                  title="RECIPES"
                  recipes={recipes}
                  handlePageChange={this.handlePageChange}
                  metadata={metadata}
                />
              )}
              {activeTab === 'categories' && (
                <RecipeItems
                  title="RECIPE CATEGORIES"
                  recipes={recipes}
                  handlePageChange={this.handlePageChange}
                  metadata={metadata}
                />
              )}
            </div>}
        </div>
      </div>);
  }
}

const mapStateToProps = state => ({
  user: state.auth.user,
  isFetching: state.isFetching,
  recipes: state.recipes.items,
  metadata: state.recipes.metadata,
});

export { FavoriteRecipes as FavoriteRecipesComponent };
export default connect(mapStateToProps)(FavoriteRecipes);
