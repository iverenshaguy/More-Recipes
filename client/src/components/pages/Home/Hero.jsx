import React from 'react';
import PropTypes from 'prop-types';
import { AddRecipeBtn } from '../../shared/Buttons';
import SearchForm from '../../shared/Forms/SearchForm';

/**
 * @exports
 * @function Hero
 * @param {object} props
 * @returns {JSX} Hero
 */
const Hero = (props) => {
  const {
    handleSearch, handleAddRecipe, handleSearchInput, searchValue
  } = props;

  return (
    <div className="container-fluid" id="hero-image">
      <div className="row mx-auto" id="hero-image-text">
        <div className="col align-self-center">
          <div className="row justify-content-center align-items-center mb-3">
            <div className="col-12">
              <h3>Search for a Recipe</h3>
            </div>
            <div className="col-8 py-4">
              <SearchForm
                size="lg"
                handleSearch={handleSearch}
                handleSearchInput={handleSearchInput}
                searchValue={searchValue}
              />
            </div>
          </div>
          <AddRecipeBtn handleClick={handleAddRecipe} />
        </div>
      </div>
    </div>);
};

Hero.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  handleAddRecipe: PropTypes.func.isRequired,
  handleSearchInput: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired
};

export default Hero;
