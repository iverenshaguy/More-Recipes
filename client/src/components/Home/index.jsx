import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Hero from './Hero';
import RecipeItems from '../shared/RecipeItems';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  /**
   * @memberof Home
   * @returns {function} updateLocationState
   */
  componentWillMount() {
    this.props.updateLocationState('home');
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero toggle={this.props.toggleModal} />
        <RecipeItems />
      </div>
    );
  }
}

Home.propTypes = {
  toggleModal: PropTypes.func.isRequired,
  updateLocationState: PropTypes.func.isRequired,
};

export default Home;
