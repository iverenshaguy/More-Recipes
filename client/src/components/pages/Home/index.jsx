import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import setCurrentLocation from '../../../actions/location';
import { fetchTopRecipes, searchRecipes } from '../../../actions/recipes';
import Hero from './Hero';
import RecipeItems from '../../shared/RecipeItems';
import PreLoader from '../../shared/PreLoader';
import { multiRecipePropTypes } from '../../../helpers/proptypes';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    isFetching: PropTypes.bool.isRequired,
    ...multiRecipePropTypes
  };

  /**
   * @constructor
   * @memberof Home
   * @returns {nothing} - Returns nothing
   */
  constructor() {
    super();

    this.state = {
      title: 'TOP RECIPES',
      currentPage: 1,
      limit: 5,
      searchValue: ''
    };

    this.handlePageChange = this.handlePageChange.bind(this);
    this.handleSearchInput = this.handleSearchInput.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  /**
   * @memberof Home
   * @return {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('home'));
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    let fetch;
    const { limit, searchValue } = this.state;

    if (this.state.title === 'TOP RECIPES') {
      fetch = fetchTopRecipes(page, limit);
    } else {
      fetch = searchRecipes(searchValue, page, limit);
    }

    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetch));
  }

  /**
   * @memberof Home
   * @param {event} event
   * @returns {state} home
   */
  handleSearchInput(event) {
    this.setState({
      searchValue: event.target.value
    });
  }

  /**
   * @memberof Home
   * @param {object} event
   * @returns {state} home
   */
  handleSearch(event) {
    event.preventDefault();

    const { searchValue, currentPage, limit } = this.state;
    this.setState({
      title: 'SEARCH RESULTS'
    }, () => this.props.dispatch(searchRecipes(searchValue, currentPage, limit)));
  }

  /**
   * @memberof Home
   * @returns {fragment} Recipe items
   */
  renderBody() {
    const { isFetching, recipes, metaData } = this.props;
    const { title } = this.state;

    if (isFetching) {
      return <PreLoader />;
    }

    return (
      <RecipeItems
        title={title}
        recipes={recipes}
        handlePageChange={this.handlePageChange}
        metaData={metaData}
      />
    );
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero
          dispatch={this.props.dispatch}
          handleSearchInput={this.handleSearchInput}
          handleSearch={this.handleSearch}
          searchValue={this.state.searchValue}
        />
        {this.renderBody()}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  recipes: state.recipes.recipes,
  metaData: state.recipes.metaData
});

export { Home as HomeComponent };

export default connect(mapStateToProps)(Home);
