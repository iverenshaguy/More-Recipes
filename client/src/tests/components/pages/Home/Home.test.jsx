import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { HomeComponent } from '../../../../components/pages/Home';

const setup = () => {
  const dispatch = jest.fn();

  const props = {
    dispatch,
    isFetching: false,
    recipes: [{ id: 1, recipeName: 'Rice', User: 'Me' }, { id: 5, recipeName: 'Beans', User: 'She' }],
    metadata: {
      firstPage: 1,
      lastPage: 3,
      page: 2,
      itemsPerPage: 5,
      pages: [1, 2, 3],
      totalCount: 13,
    }
  };

  const state = {
    title: 'TOP RECIPES',
    currentPage: 2,
    limit: 5,
    searchValue: ''
  };

  const mountedWrapper = mount(<MemoryRouter><HomeComponent {...props} /></MemoryRouter>);
  const shallowWrapper = shallow(<MemoryRouter><HomeComponent {...props} /></MemoryRouter>);

  return {
    state, props, mountedWrapper, shallowWrapper
  };
};

describe('Home', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { shallowWrapper } = setup();
    expect(toJson(shallowWrapper.find(HomeComponent))).toMatchSnapshot();
  });

  it('renders correctly when fully mounted', () => {
    const { mountedWrapper } = setup();
    expect(toJson(mountedWrapper.find(HomeComponent))).toMatchSnapshot();
    mountedWrapper.unmount();
  });

  it('renders preloader if fetching', () => {
    const { props } = setup();
    const mountedWrapper = mount(<HomeComponent {...props} isFetching />);

    expect(mountedWrapper.find('PreLoader').length).toEqual(1);
    mountedWrapper.unmount();
  });

  it('calls component will mount when mounted', () => {
    const { mountedWrapper, props } = setup();
    expect(props.dispatch).toHaveBeenCalled();
    mountedWrapper.unmount();
  });

  it('handles page change for TOP RECIPES', () => {
    const { mountedWrapper, state } = setup();

    const handlePageChangeSpy = jest.spyOn(mountedWrapper.find(HomeComponent).instance(), 'handlePageChange');

    mountedWrapper.setState(state);

    mountedWrapper.find('a.page-link').at(4).simulate('click');

    expect(handlePageChangeSpy).toHaveBeenCalled();
    expect(mountedWrapper.find(HomeComponent).instance().state.currentPage).toEqual(3);

    mountedWrapper.unmount();
  });

  it('handles page change for SEARCH RECIPES', () => {
    const { state, props } = setup();
    // rrcMock = global variable that mocks react-router context
    const mountedWrapper = mount(<HomeComponent {...props} />, rrcMock.get());

    mountedWrapper.setState({ ...state, title: 'SEARCH RESULTS', searchValue: 'rice' });
    mountedWrapper.find('a.page-link').at(3).simulate('click');

    expect(mountedWrapper.instance().state.currentPage).toEqual(2);
    mountedWrapper.unmount();
  });

  it('handles search input', () => {
    const { mountedWrapper, props, state } = setup();

    const event = { target: { value: 'Chicken' } };

    mountedWrapper.setState(state);
    mountedWrapper.find('input#search').simulate('change', event);

    expect(props.dispatch).toHaveBeenCalled();
    expect(mountedWrapper.find(HomeComponent).instance().state.searchValue).toEqual('Chicken');

    mountedWrapper.unmount();
  });

  it('handles search', () => {
    const { mountedWrapper, props, state } = setup();

    const event = { target: { value: 'Chicken' } };

    mountedWrapper.setState(state);
    mountedWrapper.find('input#search').simulate('change', event);
    mountedWrapper.find('form').simulate('submit', { preventDefault() { } });

    expect(props.dispatch).toHaveBeenCalled();
    expect(mountedWrapper.find(HomeComponent).instance().state.title).toEqual('SEARCH RESULTS');

    mountedWrapper.unmount();
  });
});
