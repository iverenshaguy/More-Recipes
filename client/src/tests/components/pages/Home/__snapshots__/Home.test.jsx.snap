// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<Home
  dispatch={[Function]}
  isFetching={false}
  metadata={
    Object {
      "firstPage": 1,
      "itemsPerPage": 5,
      "lastPage": 3,
      "page": 2,
      "pages": Array [
        1,
        2,
        3,
      ],
      "totalCount": 13,
    }
  }
  recipes={
    Array [
      Object {
        "User": "Me",
        "id": 1,
        "recipeName": "Rice",
      },
      Object {
        "User": "She",
        "id": 5,
        "recipeName": "Beans",
      },
    ]
  }
/>
`;

exports[`Home renders correctly when fully mounted 1`] = `
<Home
  dispatch={[Function]}
  isFetching={false}
  metadata={
    Object {
      "firstPage": 1,
      "itemsPerPage": 5,
      "lastPage": 3,
      "page": 2,
      "pages": Array [
        1,
        2,
        3,
      ],
      "totalCount": 13,
    }
  }
  recipes={
    Array [
      Object {
        "User": "Me",
        "id": 1,
        "recipeName": "Rice",
      },
      Object {
        "User": "She",
        "id": 5,
        "recipeName": "Beans",
      },
    ]
  }
>
  <div
    className="home"
  >
    <Hero
      dispatch={[Function]}
      handleSearch={[Function]}
      handleSearchInput={[Function]}
      searchValue=""
    >
      <div
        className="container-fluid"
        id="hero-image"
      >
        <div
          className="row mx-auto"
          id="hero-image-text"
        >
          <div
            className="col align-self-center"
          >
            <div
              className="row justify-content-center align-items-center mb-3"
            >
              <div
                className="col-12"
              >
                <h3>
                  Search for a Recipe
                </h3>
              </div>
              <div
                className="col-8 py-4"
              >
                <SearchForm
                  handleSearch={[Function]}
                  handleSearchInput={[Function]}
                  searchValue=""
                  size="lg"
                >
                  <Form
                    id="search-form"
                    onSubmit={[Function]}
                    tag="form"
                  >
                    <form
                      className=""
                      id="search-form"
                      onSubmit={[Function]}
                    >
                      <InputGroup
                        size="lg"
                        tag="div"
                      >
                        <div
                          className="input-group input-group-lg"
                        >
                          <Label
                            className="sr-only"
                            for="search"
                            tag="label"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                          >
                            <label
                              className="sr-only  form-control-label"
                              htmlFor="search"
                            >
                              Search for Recipes
                            </label>
                          </Label>
                          <Input
                            id="search"
                            onChange={[Function]}
                            placeholder="Curried Chicken Gravy..."
                            tag="p"
                            type="text"
                            value=""
                          >
                            <input
                              className="form-control"
                              id="search"
                              onChange={[Function]}
                              placeholder="Curried Chicken Gravy..."
                              type="text"
                              value=""
                            />
                          </Input>
                          <InputGroupButton
                            tag="div"
                          >
                            <div
                              className="input-group-btn"
                            >
                              <Button
                                className="search-btn"
                                color="secondary"
                                tag="button"
                                type="submit"
                              >
                                <button
                                  className="search-btn btn btn-secondary"
                                  onClick={[Function]}
                                  type="submit"
                                >
                                  <FontAwesome
                                    name="search"
                                  >
                                    <span
                                      aria-hidden={true}
                                      className="fa fa-search"
                                    />
                                  </FontAwesome>
                                </button>
                              </Button>
                            </div>
                          </InputGroupButton>
                        </div>
                      </InputGroup>
                    </form>
                  </Form>
                </SearchForm>
              </div>
            </div>
            <Button
              className="btn-default btn-lg"
              color="secondary"
              id="home-add-recipe-btn"
              onClick={[Function]}
              tag="button"
              title="New Recipe"
            >
              <button
                className="btn-default btn-lg btn btn-secondary"
                id="home-add-recipe-btn"
                onClick={[Function]}
                title="New Recipe"
                type="button"
              >
                Add a New Recipe
              </button>
            </Button>
          </div>
        </div>
      </div>
    </Hero>
    <RecipeItems
      handlePageChange={[Function]}
      metadata={
        Object {
          "firstPage": 1,
          "itemsPerPage": 5,
          "lastPage": 3,
          "page": 2,
          "pages": Array [
            1,
            2,
            3,
          ],
          "totalCount": 13,
        }
      }
      recipes={
        Array [
          Object {
            "User": "Me",
            "id": 1,
            "recipeName": "Rice",
          },
          Object {
            "User": "She",
            "id": 5,
            "recipeName": "Beans",
          },
        ]
      }
      title="TOP RECIPES"
    >
      <div
        className="container-fluid pb-5 background-grey page-margin"
        id="search-result-wrapper"
      >
        <div
          className="heading pt-4 text-center"
          id="search-result-heading"
        >
          <div
            className="col-xs-12 col-md-12 pt-3"
            id="title"
          >
            <h5>
              TOP RECIPES
            </h5>
          </div>
        </div>
        <hr />
        <div
          className="row pt-2 px-2 px-sm-3 px-md-4 px-lg-5 px-xl-5"
          id="search-result"
        >
          <RecipeCard
            key="1"
            recipe={
              Object {
                "User": "Me",
                "id": 1,
                "recipeName": "Rice",
              }
            }
          >
            <div
              className="col-xs-12 col-sm-6 col-md-4 col-lg-3 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <Link
                    replace={false}
                    to="/recipes/1"
                  >
                    <a
                      href="/recipes/1"
                      onClick={[Function]}
                    >
                      <CardHeader
                        className="py-2"
                        tag="div"
                      >
                        <div
                          className="py-2 card-header"
                        >
                          <CardText
                            className="d-inline"
                            tag="p"
                          >
                            <p
                              className="d-inline card-text"
                            >
                              <small>
                                @
                              </small>
                            </p>
                          </CardText>
                          <CardText
                            className="d-inline float-right"
                            tag="p"
                          >
                            <p
                              className="d-inline float-right card-text"
                            >
                              <small
                                className="text-muted"
                              >
                                <FontAwesome
                                  name="heart"
                                >
                                  <span
                                    aria-hidden={true}
                                    className="fa fa-heart"
                                  />
                                </FontAwesome>
                                 
                              </small>
                            </p>
                          </CardText>
                        </div>
                      </CardHeader>
                      <Card
                        className="m-0 p-0 card-image"
                        tag="div"
                      >
                        <div
                          className="m-0 p-0 card-image card"
                        >
                          <CardImg
                            alt="recipe"
                            bottom={true}
                            className="img-fluid"
                            src="images/jollof-rice-img.jpg"
                            tag="img"
                            width="100%"
                          >
                            <img
                              alt="recipe"
                              className="img-fluid card-img-bottom"
                              src="images/jollof-rice-img.jpg"
                              width="100%"
                            />
                          </CardImg>
                          <CardImgOverlay
                            className="text-center card-image-overlay text-white px-2 recipe-rating"
                            tag="div"
                          >
                            <div
                              className="text-center card-image-overlay text-white px-2 recipe-rating card-img-overlay"
                            >
                              <h5
                                className="mb-1 pt-3"
                              >
                                Rice
                              </h5>
                            </div>
                          </CardImgOverlay>
                        </div>
                      </Card>
                    </a>
                  </Link>
                </div>
              </Card>
            </div>
          </RecipeCard>
          <RecipeCard
            key="5"
            recipe={
              Object {
                "User": "She",
                "id": 5,
                "recipeName": "Beans",
              }
            }
          >
            <div
              className="col-xs-12 col-sm-6 col-md-4 col-lg-3 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <Link
                    replace={false}
                    to="/recipes/5"
                  >
                    <a
                      href="/recipes/5"
                      onClick={[Function]}
                    >
                      <CardHeader
                        className="py-2"
                        tag="div"
                      >
                        <div
                          className="py-2 card-header"
                        >
                          <CardText
                            className="d-inline"
                            tag="p"
                          >
                            <p
                              className="d-inline card-text"
                            >
                              <small>
                                @
                              </small>
                            </p>
                          </CardText>
                          <CardText
                            className="d-inline float-right"
                            tag="p"
                          >
                            <p
                              className="d-inline float-right card-text"
                            >
                              <small
                                className="text-muted"
                              >
                                <FontAwesome
                                  name="heart"
                                >
                                  <span
                                    aria-hidden={true}
                                    className="fa fa-heart"
                                  />
                                </FontAwesome>
                                 
                              </small>
                            </p>
                          </CardText>
                        </div>
                      </CardHeader>
                      <Card
                        className="m-0 p-0 card-image"
                        tag="div"
                      >
                        <div
                          className="m-0 p-0 card-image card"
                        >
                          <CardImg
                            alt="recipe"
                            bottom={true}
                            className="img-fluid"
                            src="images/jollof-rice-img.jpg"
                            tag="img"
                            width="100%"
                          >
                            <img
                              alt="recipe"
                              className="img-fluid card-img-bottom"
                              src="images/jollof-rice-img.jpg"
                              width="100%"
                            />
                          </CardImg>
                          <CardImgOverlay
                            className="text-center card-image-overlay text-white px-2 recipe-rating"
                            tag="div"
                          >
                            <div
                              className="text-center card-image-overlay text-white px-2 recipe-rating card-img-overlay"
                            >
                              <h5
                                className="mb-1 pt-3"
                              >
                                Beans
                              </h5>
                            </div>
                          </CardImgOverlay>
                        </div>
                      </Card>
                    </a>
                  </Link>
                </div>
              </Card>
            </div>
          </RecipeCard>
        </div>
        <Pagination
          items={
            Array [
              Object {
                "User": "Me",
                "id": 1,
                "recipeName": "Rice",
              },
              Object {
                "User": "She",
                "id": 5,
                "recipeName": "Beans",
              },
            ]
          }
          metadata={
            Object {
              "firstPage": 1,
              "itemsPerPage": 5,
              "lastPage": 3,
              "page": 2,
              "pages": Array [
                1,
                2,
                3,
              ],
              "totalCount": 13,
            }
          }
          onPageChange={[Function]}
        >
          <div
            className="mt-5"
          >
            <nav
              aria-label="Page navigation"
            >
              <Pagination
                className="justify-content-center"
                tag="ul"
              >
                <ul
                  className="justify-content-center pagination"
                >
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          First
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          <span
                            aria-hidden="true"
                          >
                            «
                          </span>
                          <span
                            className="sr-only"
                          >
                            Previous
                          </span>
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    key="1"
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          1
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className="active"
                    key="2"
                    tag="li"
                  >
                    <li
                      className="active page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          2
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    key="3"
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          3
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          <span
                            aria-hidden="true"
                          >
                            »
                          </span>
                          <span
                            className="sr-only"
                          >
                            Next
                          </span>
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          Last
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                </ul>
              </Pagination>
            </nav>
          </div>
        </Pagination>
      </div>
    </RecipeItems>
  </div>
</Home>
`;
